policy/pa.go

```diff
@@ -285,3 +286,28 @@ func (pa *AuthorityImpl) ChallengesFor(identifier core.AcmeIdentifier) ([]core.C
 
 	return shuffled, shuffledCombos
 }
+
+// ExtractDomainIANASuffix returns the public suffix of the domain using only the "ICANN"
+// section of the Public Suffix List database.
+// If the domain does not end in a suffix that belongs to an IANA-assigned
+// domain, ExtractDomainIANASuffix returns an error.
+func extractDomainIANASuffix(name string) (string, error) {
+	if name == "" {
+		return "", fmt.Errorf("Blank name argument passed to ExtractDomainIANASuffix")
```

Comment:
Actually, there is no such test case currently defined in the PSL, and my library currently returns an empty string (as a result of "*" being the default rule). I've added that check to keep compatibility with the existing implementation.

There is a test case, it's covered by https://github.com/letsencrypt/boulder/pull/1969/files#diff-4926eb5faa4f5857f78ea784d9176239R260

I'll investigate if it makes sense to handle this error at the library level.